noir/aztec_macros/src/lib.rs

```diff
@@ -1601,3 +1601,143 @@ fn event_signature(event: &StructType) -> String {
     let fields = vecmap(event.get_fields(&[]), |(_, typ)| signature_of_type(&typ));
     format!("{}({})", event.name.0.contents, fields.join(","))
 }
+
+fn inject_compute_note_hash_and_nullifier(
+    crate_id: &CrateId,
+    context: &mut HirContext,
+    unresolved_traits_impls: &Vec<UnresolvedTraitImpl>,
+    collected_functions: &mut Vec<UnresolvedFunctions>,
+) -> Result<(), (MacroError, FileId)> {
+    // We first fetch modules in this crate which correspond to contracts, along with their file id.
+    let mut contract_module_file_ids: Vec<(LocalModuleId, FileId)> = context.def_map(crate_id).unwrap().modules().iter()
```

Comment:
All these usages of `unwrap()` should be switched to `expect("..")` with some message or transformed into MacroErrors.